calculator/parser.py

```diff
@@ -0,0 +1,132 @@
+BINARY_OPERATORS = ["+", "-", "*", "/"]
+
+
+def binary_operator(num_1, num_2, op):
+    """
+    Evaluates and acts on different binary operators
+
+    Parameters
+    ----------
+    num_1 : integer
+        The first number to be operated on
+    
+    num_2 : integer
+        The second number to be operated on
+    
+    op : char
+        The operator that is going to be applied to the numbers
+
+    Returns
+    -------
+    integer : The final answer after after the nums are operated on
+    """
+
+    if op == "+":
+        return num_1 + num_2
+
+    elif op == "-":
+        return num_2 - num_1
+
+    elif op == "*":
+        return num_1 * num_2
+
+    elif op == "/":
+        return num_2 / num_1
+
+
+def push(e, stack):
+    """
+    Pushes the element (e) to the given stack
+
+    Parameters
+    ----------
+    e : element
+        The element to be pushed on top of the stack
+
+    stack : list
+        The stack that e is going to be pushed onto
+
+    Returns
+    -------
+    list : The stack that is now updated to have e pushed ontop of it
+    """
+    return stack.append(e)
+
+
+def pop(stack):
+    """
+    Retrieves the last element from the given stack and deletes it
+
+    Parameters
+    ----------
+    stack : list
+        The stack (in list form) to be operated on
+
+    Returns
+    -------
+    stack : The popped stack
+    """
+    return stack.pop()
+
+
+def is_numeric_token(token):
+    """
+    Says whether or not the token input is a number
+
+    Parameter
+    ---------
+    token : token
+        The token to be evalueated
+
+    Returns
+    -------
+    bool: Whether or not this token is a numeric character
+    """
+
+    if type(token) == float or type(token) == int:
+        return True
+
+    else:
+        return False
+
+
+def rpn(tokens):
+    """
+    Parses RPN and returns an answer
+
+    Parameters
+    ----------
+    tokens : list
+        list of tokens
+
+    Returns
+    -------
+    rpn_stack : list containing 1 element (the answer)
```

Comment:
Please just return the answer, not the stack.  The reason why you wouldn't do that is that you're exposing HOW the algorithm is implemented (i.e. with a stack) to the caller, when the caller shouldn't care.  If, for instance, you decide to use another approach (that may be faster or whatever), then the caller shouldn't know or care about the underlying implementation, just that the function parses a token list in RPN order.